fix(my-account): new payment method style tweaks

miguelpeixe opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

NPPD-485

Style improvements to the new payment method form:

| Before | After |
| --- | --- |
| <img width="679" height="757" alt="image" src="https://github.com/user-attachments/assets/cd805e18-7c97-4939-96d4-7716b78096cd" /> | <img width="688" height="779" alt="image" src="https://github.com/user-attachments/assets/97b63e76-cd11-40ea-840a-2d9e07bba6b6" /> |

### How to test the changes in this Pull Request:

Add a new payment method via My Account -> Payment information and confirm it works as expected

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

miguelpeixe committed:
fix(my-account): new payment method style tweaks

thomasguillot committed:
feat(newspack-ui): if only one accordion, remove background, border, padding

thomasguillot committed:
fix: radio/checkbox gaps

thomasguillot commented:
@miguelpeixe thank you for this PR. These are details that bothered me for quite some time now. I went even further with the  only 1 payment method:

<img width="1572" height="1568" alt="" src="https://github.com/user-attachments/assets/6afd7ac4-f345-40be-ab00-62723841b96f" />

I wonder if we can do something similar when we're checking out

<img width="2536" height="1626" alt="" src="https://github.com/user-attachments/assets/d0571024-e2e3-4b3c-9296-c28ee1b13477" />

miguelpeixe commented:
Awesome! I like that better too.

If you look at NPPD-485 you'll see that the "Change Payment Method" had this approach as well, but at some point got boxed, not sure at which point.

Let's use this PR to fix the inconsistencies

miguelpeixe commented:
This is how it looks now, btw

<img width="684" height="561" alt="image" src="https://github.com/user-attachments/assets/3823486c-01c4-4fe3-9ed5-d1fa202b7fad" />

thomasguillot committed:
fix: checkbox font size

miguelpeixe committed:
Merge branch 'trunk' into fix/my-account-new-payment-method-style-tweaks

dkoo reviewed (changes requested):
Styles look good!

<img width="681" height="685" alt="Screenshot 2025-10-10 at 11 15 57 AM" src="https://github.com/user-attachments/assets/6995dc71-c5f3-43e4-975a-16805fd7a350" />

I'm seeing a PHP warning on the Payment Information page:

```
PHP Warning:  Undefined array key "form_id" in /newspack-repos/newspack-plugin/includes/class-newspack-ui.php on line 219
```

Also, this might be for another PR, but is it possible to make both checkboxes at the bottom of the modal either checked by default, or even checked by default + visually hidden?

miguelpeixe committed:
fix: add form_id

miguelpeixe committed:
fix: hide "Save payment method" checkbox when adding new

miguelpeixe commented:
Thank you for the review!

> I'm seeing a PHP warning on the Payment Information page:
> PHP Warning:  Undefined array key "form_id" in /newspack-repos/newspack-plugin/includes/class-newspack-ui.php on line 219

Good catch! This is in trunk too, from #4216. Fixed in 2edb7ce016752f9a25d3684f4e6721bfb332aa20

>  Also, this might be for another PR, but is it possible to make both checkboxes at the bottom of the modal either checked by default, or even checked by default + visually hidden?

I agree that the first checkbox shouldn't exist, and checking it doesn't really change anything in this form. Fixed in f0f8016eedeee7345023b23dcbad5c97d71dc9d4.

IMO, the second checkbox is relevant and shouldn't be checked by default.

dkoo commented:
> IMO, the second checkbox is relevant and shouldn't be checked by default.

I'm open to discussing this further, but it's how things work in v0 now. #2934 and #3643 made the checkbox checked by default and hid it visually (because publishers were receiving feedback from readers being confused by it and/or not checking it after adding a new payment method, resulting in failed renewals), so showing it now would be a change in behavior.

miguelpeixe commented:
Ah! Thank you for the additional context. I didn't realize that. I'll fix that then

miguelpeixe commented:
I also realized that what was done in #2934 no longer works. Maybe something from https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3461

Investigating

miguelpeixe committed:
fix: allow 'checked' input attr

miguelpeixe committed:
fix: hook caller

miguelpeixe committed:
fix: hide checked field

miguelpeixe commented:
Took me a while... it turned out to be kses filtering out the `checked` attribute. Fixed in fe16aae39be0d99900793f49cf6abddeccbaa426

Should be good now!
